types/block_test.go

```diff
@@ -113,6 +113,44 @@ func TestBlockID(t *testing.T) {
 	}
 }
 
+// TestHeaderID probes the ID function of the BlockHeader type.
+func TestHeaderID(t *testing.T) {
```

Comment:
actually, I see now that the definition of Block.ID is `crypto.HashObject(b.Header())`

sanity check is probably not needed